modules/mesaides/templates/nginx_config.erb

```diff
@@ -0,0 +1,56 @@
+# MANAGED BY PUPPET
+server {
+  listen                *:80 default_server;
+  server_name           <%= @name %>;
+
+  access_log            /var/log/nginx/<%= @name %>.access.log combined;
+  error_log             /var/log/nginx/<%= @name %>.error.log;
+
+  location /.well-known {
+    root                <%= @webroot_path %>;
+  }
+<% if @use_ssl and File.exists?("/etc/letsencrypt/live/#{@name}/fullchain.pem") -%>
```

Comment:
Well, that is a trick to prevent to many configuration items. There are 3 steps:
- Basic - NO HTTPS
- You want to serve using HTTPS (because `/opt/mes-aides/use_ssl` exists) but you don't have an SSL certificate
- You have an SSL certificate and can serve HTTPS

I use `/opt/mes-aides/use_ssl` to trigger a certificate request (2) and rely on `/etc/letsencrypt/live/#{@name}/fullchain.pem` presence to make sure I want to serve HTTPS (3).

At some point, 2 and 3 could not happen in the same puppet run but the addition of https://github.com/sgmap/mes-aides-ops/pull/19/files#diff-eb7229d4d00546ca719fcff7588b8e9fR31 should make it possible.